docs/en/framework/scopes.md

```diff
@@ -1,50 +1,208 @@
+$binder->bind(Interface::class, factory(...));
+```
+
+#### Scope destroying
+
+Once a scope is finished (destroyed), all the dependencies (including singletons) that were initialized within that
+scope are also destroyed. This ensures clean and efficient resource management. The next time a scope with the same name
+is created, these bindings will be resolved anew, following the configurations you've set using the `BinderInterface`.
+
+#### Overriding Default Bindings
+
+When you use the `Container::runScoped()` method to create a new scope, you can optionally pass the `$bindings` argument
+to override the default bindings for that specific scope.
+
+- These custom bindings will take precedence within the newly created scope.
+- This won't affect the default bindings that you've previously defined for that named scope.
+
+**Example:**
+
+```php
+$container->bindSingleton(SomeInterface::class, SomeImplementation::class);
+
+$container->runScoped(closure: function ($container) {
+    $instance = $container->get(SomeInterface::class);
+    // Your code here
+}, bindings: [SomeInterface::class => AnotherImplementation::class], name: 'request');
+```
+
+In this example, even though the `request` scope might have a default binding for `SomeInterface::class`, this specific
+run of the scope is using A`notherImplementation::class`.
+
+### Scope Destroying
+
+When a dependency is resolved within a scope, and that scope is destroyed, the finalize method specified by this
+attribute will be called. The purpose of the finalize method is to perform any necessary cleanup or finalization actions
+before the scope is destroyed. In this case you can use `Finalize` attribute to control finalizing process. The
+attribute provides a convenient way to handle resource cleanup and ensure proper destruction of objects when a scope is
+no longer needed.
+
+```php
+#[Finalize(method: 'finalize')]
+class Foo
+{
+    public bool $finalized = false;
+
+    public function finalize(Logger $logger): void
+    {
+        $this->finalized = true;
+        $logger->log();
+    }
+}
+```
+
+### Naming a Scope
+
+You can optionally give a name to the scope with the `$name` parameter. This can be useful for tracking or managing
+specific scopes.
+
+**Restrictions**
+
+These restrictions ensure proper usage and prevent conflicts within the scope hierarchy. Here are the key restrictions:
+
+1. **Named Scope Uniqueness:** Named scopes within the same hierarchy must have unique names. This means that you cannot
+   have multiple scopes with the same name at the same level of the scope hierarchy. Each named scope should have a
+   distinct and meaningful name to avoid any ambiguity.
+
+2. **Default Bindings for Named Scopes:** Named scopes can have default bindings associated with them. It's important to
+   note that default bindings do not affect already created container instances, except for the root container. Default
```

Comment:
```suggestion
   note that changing of default bindings does not affect already created container instances with the same scope name, except for the root container. Default
```

Reply:
I think the restriction is not a restriction. It's a behavior of the container binder.